metricflow/validation/data_warehouse_model_validator.py

```diff
@@ -238,7 +238,7 @@ def gen_dimension_tasks(
                             ),
                             element_type=SemanticModelElementType.DIMENSION,
                         ),
-                        error_message=f"Unable to query dimension `{spec.element_name}` on semantic model `{semantic_model.name}` in data warehouse",
+                        error_message=f"Unable to query dimension `{spec.qualified_name}` on semantic model `{semantic_model.name}` in data warehouse",
```

Comment:
Q: This was logic already here, but do we need to query against every grain for a time dimension? Could we just consolidate this to only query for the time dimension column or are there cases where the grain/dow casting fails

Reply:
Hmmmm cases I can think of:
- the grain is too small for what the time dimension supports
- date part is not supported for cumulative metrics

Reply:
So since these arent metric queries and just straight `SELECT <dim column> FROM sem_model` do those still apply?

Reply:
If the user can request a grain then the grain one still applies! Like is there a date trunc happening here? If so it could fail if the time column can't be trunced to a certain level

Reply:
Actually same for date part since they could request a date part that's too small for the column

Reply:
Yea the date trunc happens here